### DISCUSSION TITLE
I got the following error when "make PythonAPI":   make: *** [Util/BuildTools/Linux.mk:137: setup] Error 1

### DISCUSSION THREAD

**Discussion by Moneyboy63211** [carla-simulator/carla#5856] [Q&A]
I got the following error when "make PythonAPI":

"clang++" -c -x c++ -fvisibility-inlines-hidden -fPIC -m64 -pthread -O3 -Wall -fvisibility=hidden -Wno-inline -fPIC -std=c++14 -DBOOST_ERROR_CODE_HEADER_ONLY -DBOOST_ALL_NO_LIB=1 -DBOOST_NUMPY_SOURCE -DNDEBUG -I"." -I"/home/tianle/venv_carla/.venv/include/python3.7" -I"/home/tianle/venv_carla/.venv/lib/python3.7/site-packages/numpy/core/include" -o "bin.v2/libs/python/build/clang-linux-10.0/release/python-3.7/threading-multi/visibility-hidden/numpy/ndarray.o" "libs/python/src/numpy/ndarray.cpp"

...failed clang-linux.compile.c++.without-pch bin.v2/libs/python/build/clang-linux-10.0/release/python-3.7/threading-multi/visibility-hidden/numpy/ndarray.o...
clang-linux.compile.c++.without-pch bin.v2/libs/python/build/clang-linux-10.0/release/python-3.7/threading-multi/visibility-hidden/numpy/ufunc.o
In file included from libs/python/src/numpy/ufunc.cpp:8:
In file included from ./boost/python/numpy/internal.hpp:17:
In file included from ./boost/python.hpp:11:
In file included from ./boost/python/args.hpp:8:
In file included from ./boost/python/detail/prefix.hpp:13:
./boost/python/detail/wrap_python.hpp:57:11: fatal error: 'pyconfig.h' file not found

include <pyconfig.h>
      ^~~~~~~~~~~~
1 error generated.

"clang++" -c -x c++ -fvisibility-inlines-hidden -fPIC -m64 -pthread -O3 -Wall -fvisibility=hidden -Wno-inline -fPIC -std=c++14 -DBOOST_ERROR_CODE_HEADER_ONLY -DBOOST_ALL_NO_LIB=1 -DBOOST_NUMPY_SOURCE -DNDEBUG -I"." -I"/home/tianle/venv_carla/.venv/include/python3.7" -I"/home/tianle/venv_carla/.venv/lib/python3.7/site-packages/numpy/core/include" -o "bin.v2/libs/python/build/clang-linux-10.0/release/python-3.7/threading-multi/visibility-hidden/numpy/ufunc.o" "libs/python/src/numpy/ufunc.cpp"

...failed clang-linux.compile.c++.without-pch bin.v2/libs/python/build/clang-linux-10.0/release/python-3.7/threading-multi/visibility-hidden/numpy/ufunc.o...
clang-linux.compile.c++.without-pch bin.v2/libs/python/build/clang-linux-10.0/release/python-3.7/threading-multi/visibility-hidden/numpy/scalars.o
In file included from libs/python/src/numpy/scalars.cpp:8:
In file included from ./boost/python/numpy/internal.hpp:17:
In file included from ./boost/python.hpp:11:
In file included from ./boost/python/args.hpp:8:
In file included from ./boost/python/detail/prefix.hpp:13:
./boost/python/detail/wrap_python.hpp:57:11: fatal error: 'pyconfig.h' file not found

include <pyconfig.h>
      ^~~~~~~~~~~~
1 error generated.

"clang++" -c -x c++ -fvisibility-inlines-hidden -fPIC -m64 -pthread -O3 -Wall -fvisibility=hidden -Wno-inline -fPIC -std=c++14 -DBOOST_ERROR_CODE_HEADER_ONLY -DBOOST_ALL_NO_LIB=1 -DBOOST_NUMPY_SOURCE -DNDEBUG -I"." -I"/home/tianle/venv_carla/.venv/include/python3.7" -I"/home/tianle/venv_carla/.venv/lib/python3.7/site-packages/numpy/core/include" -o "bin.v2/libs/python/build/clang-linux-10.0/release/python-3.7/threading-multi/visibility-hidden/numpy/scalars.o" "libs/python/src/numpy/scalars.cpp"

...failed clang-linux.compile.c++.without-pch bin.v2/libs/python/build/clang-linux-10.0/release/python-3.7/threading-multi/visibility-hidden/numpy/scalars.o...
clang-linux.compile.c++.without-pch bin.v2/libs/python/build/clang-linux-10.0/release/python-3.7/threading-multi/visibility-hidden/numpy/matrix.o
In file included from libs/python/src/numpy/matrix.cpp:8:
In file included from ./boost/python/numpy/internal.hpp:17:
In file included from ./boost/python.hpp:11:
In file included from ./boost/python/args.hpp:8:
In file included from ./boost/python/detail/prefix.hpp:13:
./boost/python/detail/wrap_python.hpp:57:11: fatal error: 'pyconfig.h' file not found

include <pyconfig.h>
      ^~~~~~~~~~~~
1 error generated.

..
..
..
.
.

ons-variant-shared.cmake
common.copy /home/tianle/Documents/carla/Build/boost-1.72.0-c8-source/stage/lib/libboost_program_options.so.1.72.0
ln-UNIX /home/tianle/Documents/carla/Build/boost-1.72.0-c8-source/stage/lib/libboost_program_options.so
ln-UNIX /home/tianle/Documents/carla/Build/boost-1.72.0-c8-source/stage/lib/libboost_program_options.so.1
ln-UNIX /home/tianle/Documents/carla/Build/boost-1.72.0-c8-source/stage/lib/libboost_program_options.so.1.72
common.copy /home/tianle/Documents/carla/Build/boost-1.72.0-c8-source/stage/lib/cmake/boost_program_options-1.72.0/libboost_program_options-variant-shared.cmake
...failed updating 66 targets...
...skipped 24 targets...
...updated 173 targets...
make: *** [Util/BuildTools/Linux.mk:137: setup] Error 1

**Comment by timolohrenz** [comment]
I remember that I had a this error too and I think it was due to an anaconda environment I had accidentally activated during compilation.

**Comment by LJL0511** [comment]
modify the link in the file  SetUp.sh . （XERCESC_VERSION=3.2.4）. also modify the file BuildOSM2ODR.sh(line:152) xereces-c-3.2.4.

**Reply by vihaancp** [reply]
not working unfortunately. line 152 for me doesnt exist. can you help?

**Comment by HyeontaeSung** [comment]
I have the same issue. Have you solved it?
